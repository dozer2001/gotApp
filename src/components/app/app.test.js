import React from 'react';
import {shallow} from 'enzyme';
import App from './app';


describe('Testing <App/>', () => {
    const app = shallow(<App/>);
    describe('Testing snap & state', () => {
        it('app have rendered correctly', () => {
            expect(app).toMatchSnapshot();
        });
        it('app state "app" is empty obj', () => {
            expect(app.state().dis).toBeTruthy();
        });
        it('app state "error" is ', () =>{
            expect(app.state().error).toBeFalsy();
        });

    });
});