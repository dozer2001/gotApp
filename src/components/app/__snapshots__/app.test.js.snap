// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <App/> Testing snap & state app have rendered correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ForwardRef>
        <Container
          tag="div"
        >
          <Header />
        </Container>
        <Container
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              lg={
                Object {
                  "offset": 0,
                  "size": 5,
                }
              }
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <RandomChar />
              <button
                onClick={[Function]}
              >
                Klick me
              </button>
            </Col>
          </Row>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/characters"
          />
          <Route
            component={[Function]}
            path="/houses"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/books"
          />
          <Route
            path="/books/:id"
            render={[Function]}
          />
        </Container>
      </ForwardRef>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Container
            tag="div"
          >
            <Header />
          </Container>,
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 0,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <RandomChar />
                <button
                  onClick={[Function]}
                >
                  Klick me
                </button>
              </Col>
            </Row>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/characters"
            />
            <Route
              component={[Function]}
              path="/houses"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/books"
            />
            <Route
              path="/books/:id"
              render={[Function]}
            />
          </Container>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Header />,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <RandomChar />
                  <button
                    onClick={[Function]}
                  >
                    Klick me
                  </button>
                </Col>
              </Row>,
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/characters"
              />,
              <Route
                component={[Function]}
                path="/houses"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/books"
              />,
              <Route
                path="/books/:id"
                render={[Function]}
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <RandomChar />
                  <button
                    onClick={[Function]}
                  >
                    Klick me
                  </button>
                </Col>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <RandomChar />,
                    <button
                      onClick={[Function]}
                    >
                      Klick me
                    </button>,
                  ],
                  "lg": Object {
                    "offset": 0,
                    "size": 5,
                  },
                  "tag": "div",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Klick me",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Klick me",
                    "type": "button",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/characters",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/houses",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/books",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/books/:id",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gZMcBi",
          "isStatic": true,
          "rules": Array [
            "
background-image: url(",
            "got.jpeg",
            ")
    overflow-x: hidden;
    background: url('img/got.jpeg') center center no-repeat;
    background-size: cover;
    font-size: 16px;
    height: 100%;	

",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-gZMcBi",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef>
          <Container
            tag="div"
          >
            <Header />
          </Container>
          <Container
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                lg={
                  Object {
                    "offset": 0,
                    "size": 5,
                  }
                }
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <RandomChar />
                <button
                  onClick={[Function]}
                >
                  Klick me
                </button>
              </Col>
            </Row>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/characters"
            />
            <Route
              component={[Function]}
              path="/houses"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/books"
            />
            <Route
              path="/books/:id"
              render={[Function]}
            />
          </Container>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Container
              tag="div"
            >
              <Header />
            </Container>,
            <Container
              tag="div"
            >
              <Row
                tag="div"
              >
                <Col
                  lg={
                    Object {
                      "offset": 0,
                      "size": 5,
                    }
                  }
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <RandomChar />
                  <button
                    onClick={[Function]}
                  >
                    Klick me
                  </button>
                </Col>
              </Row>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                path="/characters"
              />
              <Route
                component={[Function]}
                path="/houses"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/books"
              />
              <Route
                path="/books/:id"
                render={[Function]}
              />
            </Container>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Header />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Row
                  tag="div"
                >
                  <Col
                    lg={
                      Object {
                        "offset": 0,
                        "size": 5,
                      }
                    }
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <RandomChar />
                    <button
                      onClick={[Function]}
                    >
                      Klick me
                    </button>
                  </Col>
                </Row>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  path="/characters"
                />,
                <Route
                  component={[Function]}
                  path="/houses"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/books"
                />,
                <Route
                  path="/books/:id"
                  render={[Function]}
                />,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <Col
                    lg={
                      Object {
                        "offset": 0,
                        "size": 5,
                      }
                    }
                    tag="div"
                    widths={
                      Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                    }
                  >
                    <RandomChar />
                    <button
                      onClick={[Function]}
                    >
                      Klick me
                    </button>
                  </Col>,
                  "tag": "div",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <RandomChar />,
                      <button
                        onClick={[Function]}
                      >
                        Klick me
                      </button>,
                    ],
                    "lg": Object {
                      "offset": 0,
                      "size": 5,
                    },
                    "tag": "div",
                    "widths": Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Klick me",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Klick me",
                      "type": "button",
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/characters",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/houses",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/books",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "path": "/books/:id",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gZMcBi",
            "isStatic": true,
            "rules": Array [
              "
background-image: url(",
              "got.jpeg",
              ")
    overflow-x: hidden;
    background: url('img/got.jpeg') center center no-repeat;
    background-size: cover;
    font-size: 16px;
    height: 100%;	

",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-gZMcBi",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
